sdk/python/kubeflow/training/api/training_client.py

```diff
@@ -81,6 +91,169 @@ def __init__(
             )
         self.job_kind = job_kind
 
+    def train(
+        self,
+        name: str = None,
+        namespace: str = None,
+        num_workers: int = 1,
+        num_procs_per_worker: int = 1,
+        storage_config: Dict[Literal["size", "storage_class"], str] = {"size": "10Gi"},
+        model_provider_parameters: HuggingFaceModelParams = None,
+        dataset_provider_parameters: Union[HfDatasetParams, S3DatasetParams] = None,
+        train_parameters: HuggingFaceTrainParams = None,
+        resources_per_worker: Union[dict, client.V1ResourceRequirements, None] = {
+            "cpu": 1,
+            "memory": "2Gi",
+        },
```

Comment:
Let's remove default resources similar to Katib: https://github.com/kubeflow/katib/blob/master/sdk/python/v1beta1/kubeflow/katib/api/katib_client.py#L153, usually it's up to the user to set those values.